docs/hyperopt.md

```diff
@@ -144,25 +144,27 @@ it will end with telling you which paramter combination produced the best profit
 
 The search for best parameters starts with a few random combinations and then uses a
 regressor algorithm (currently ExtraTreesRegressor) to quickly find a parameter combination
-that minimizes the value of the objective function `calculate_loss` in `hyperopt.py`.
+that minimizes the value of the [loss function](#loss-functions).
 
 The above setup expects to find ADX, RSI and Bollinger Bands in the populated indicators.
 When you want to test an indicator that isn't used by the bot currently, remember to
 add it to the `populate_indicators()` method in `hyperopt.py`.
 
 ## Loss-functions
 
-Each hyperparameter tuning requires a target. This is usually defined as a loss function, which get's closer to 0 for increasing values.
+Each hyperparameter tuning requires a target. This is usually defined as a loss function (sometimes also called objective function), which should decrease for more desirable results, and increase for bad results.
 
-FreqTrade uses a default loss function, which has been with freqtrade since the beginning and optimizes mostly for short trade duration and avoiding losses.
+By default, FreqTrade uses a loss function, which has been with freqtrade since the beginning and optimizes mostly for short trade duration and avoiding losses.
 
-A different version this can be used by using the `--hyperopt-loss <Class-name>` argument.
+A different version this can be used by using the `--hyperopt-loss-class <Class-name>` argument.
```

Comment:
Wait a minute please, seems I missed it, sorry...

Is this option actually called `--hyperopt-loss-class` now? In this case the `--strategy` option can also be called `--strategy-class`... Let's rename it (as minimum) to `--hyperopt-loss` (which sounds so-so, returning back to our discussion on the meaning of Loss) or (a compromize and the current proposal after I found I missed this point) -- let's name it `--hyperopt-objective`, describe in the configuration.md and in the option help string that it defines 'the name of the class with the hyperopt objective loss function' or something like that, and leave all the '_loss' and 'Loss' in the code in the names of the vars, funcs and classes...

What I don't like here is `-class` in the name of the option...

Reply:
well the nice thing about argparse is that you can pass `--hyperopt-loss` and it'll work just fine until it finds a conflict (so --hyperopt will not work anymore, but `--hyperopt-l` should.

I thought being explicit (it's a class we expect, not a filename) would help - but if you see it differently i can remove this.

calling the option `hyperopt-objective` - but everything else is "loss" does make no sense to me, and is wildly confusing.